### DISCUSSION TITLE
Coloring a dimplot with my own color vector

### DISCUSSION THREAD

**Discussion by ghost** [satijalab/seurat#4902] [Q&A]
Hello,

I would like to color all cells in a DimPlot with a precomputed color vector. Is there any easy way to do so? My problem is that the data has been clustered. Therefore, using cols=myVector will use the first entries of the vector as colors for the clusters. 
Is there a way to tell DimPlot to ignore group.by?
Alternatively, how can we tell DimPlot to group.by cell name? (effectively creating as many classes as cells)

Thank you

**Comment by samuel-marsh** [comment]
Hi,

So if your intent is to color the cells individually then yes you can just create `@meta.data` column that contains the cell names and then use `group.by` on that column (Don't forget to add `+ NoLegend()` too).

```
pbmc[["cell_names"]] <- colnames(pbmc)
DimPlot(pbmc, group.by = "cell_names") + NoLegend()
```

Best,
Sam

**Reply by ghost** [reply]
Sorry for the delay of reply. I think I was not clear in my enquiry. The group.by sent you off track.

I want to color the cells with precomputed values. For instance, the relative normalized expression of two genes (but that could be anything). For that, I can create my color vector as:

# Creates a vector of relative counts between two genes
counts <- as.matrix(GetAssayData(object = obj, assay = "RNA", slot = "counts"))
MyGenes <- c("XXX","YYY")
maxGene1<-max(counts[MyGenes[1],])
maxGene2<-max(counts[MyGenes[2],])
relCounts<-counts[MyGenes[1],]/maxGene1-counts[MyGenes[2],]/maxGene2

# Build a color vector based on relative counts
rbPal <- colorRampPalette(c('darkred',"yellow",'darkcyan'))
datcol <- rbPal(10)[as.numeric(cut(relCounts,breaks = 10))]

> length(colnames(obj))
[1] 7707
> length(datcol)
[1] 7707

Now, I want to use that color vector to color all cells according to their relative expression. Again, the later detail is just that. what matters is coloring a cell in the vector of cells with the corresponding color in the vector of colors. The following does not work:

DimPlot(obj,reduction="tsne", group.by = "cell_names") + NoLegend()+ scale_color_manual(values = datcol)

The plot should  look like what we get with "features = MyGenes, blend=TRUE, combine = FALSE", but it does not.

**Reply by timoast** [reply]
If you want a continuous color scale, you should use `FeaturePlot()` rather than `DimPlot()`. `FeaturePlot()` can plot any continuous values stored in the object metadata, and then you can apply your own colorscale using standard `ggplot2` functions
